### PR TITLE
Speargun: Now actually a gun! Again!

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Turns the speargun from the syndikit bundle into a bolt action rifle instead of a pnuematic cannon.

The spears the rifle fires embeds using the new bullet embedding system by Ryll. The spears do a lot of fucking damage.

The new bundle includes:

**The Ballistic Harpoon Gun**

The weapon is a single shot rifle that can be loaded with harpoons. The rifle, being bolt action, requires pulling back the bolt, loading the new harpoon, then locking the bolt every time you wish to fire the weapon. This is more busy work than the pnuematic cannon version, but more reliable in functionality without any messy business with throwing.

The harpoons do 60 force, have a high wound chance against unarmored targets and very high embed chance, with a 100% against unarmored targets. They also have considerable armor piercing at 50, letting you nearly entirely shoot through a ballistic vest.

The weapon is essentially a powered-up single shot mosin or a significantly more powerful improvised shotgun filled with slugs.

**The Quiver**

Now a bag that goes on your belt. It holds up to 40 harpoons for your harpoon gun. This change was done because the harpoons cannot be retrieved, unlike the magspears, which were simply hacky ninja stars and could be pulled out with surgical tools/wirecutters. Previously you had 20 magspears, which because of their retrievable nature lasted a little longer than you would expect.

**Sprites**

New back sprites for the harpoon gun, ensuring nobody is missing it when you have it. Additionally, the Mosin Nagant now has it's back sprite as a suit storage sprite.

New sprite for the quiver.

## Why It's Good For The Game

Denton, when he originally reworked magspears back here (https://github.com/tgstation/tgstation/pull/42102) into a pneumatic cannon did so because they weren't able to embed as a bullet. Now, bullets embed, so it's back to being a gun again!

(Also adds some sproites)
~~(this will conflict with my own pr aaaah)~~(it conflicted but I feex)

## Changelog
:cl:
tweak: Reworks the speargun BACK to being a gun! Complete with embedding harpoons! How exciting.
imageadd: Adds new mob sprites for the harpoon gun and a suit storage sprite for the mosin nagant. Also adds a new quiver sprite.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile/bullets/rifle.dm [hunk 1]
```diff
@@ -25,3 +25,15 @@
 	name = "enchanted 7.62 bullet"
 	damage = 20
 	stamina = 80
+
+// Harpoons (Harpoon Gun)
+
+/obj/projectile/bullet/harpoon
+	name = "harpoon"
+	icon_state = "gauss"
+	damage = 80
```
Comment (Fikou) [inline code/modules/projectiles/projectile/bullets/rifle.dm:34]: 👀

**Comment by necromanceranne** [comment]
I went and checked out the magspear gun again since I believe the last time I ever was shot by it was pre-scarp rework, and it did an insane amount of damage at the time because embedded objects did insane amounts of damage in general.

So here a weak visual difference between the current and new:

test 1 speargun

test 2 speargun

I'd appreciate feedback on whether this is too much or not. Do note that the new spear gun requires unbolting the gun, loading the gun, then locking the bolt before you can fire, so there is a greater amount of actions between shots.

**Comment by Pumpkinoe** [comment]
lookin' awesome, dang
I feel like the buff is a good change: it's a pretty rare weapon so giving it a little bit of an extra oomph is a perfect decision, especially considering it did less than what a revolver can do with a single .357 round that you can make from an autolathe
so, yeah, speargun 4 da win

**Comment by necromanceranne** [comment]
This is now conflicting with my own pr, everyone point at laugh at me. (not too much, I'm sensitive)

**Comment by Pumpkinoe** [comment]
image

**Comment by wesoda25** [comment]
Fun but way too much. Shouldn't do over 50 damage imo.

**Comment by necromanceranne** [comment]
Is there any particular reason for 50? The weapons rubbing shoulders with revolver boxes and sniper rifles, and free deswords and spells. The kits tend to be fairly powerful, so a one bullet at a time rifle is somewhat tame all things considered, even if it is 20 tc for what is literally just this rifle. 

Now that I think about it, most kits are either utterly useless for anything but one and done gimmicks, or just straight up murder tools on the cheap. One or two are even perfectly designed for rushing through murdering people with considerable efficiency, like meta op.

**Comment by Pumpkinoe** [comment]
80 is still perfect in my opinion, agreeing with Necro here. You only get a limited amount of spears so it makes sense that they're a power house, considering the spear gun is arguably rarer and less efficient than a revolver currently (And you can only get the speargun out of RNG & spending 20 TC.)  Keep it at 80!

**Comment by necromanceranne** [comment]
Actually I'm gonna go ask Denton about what he intended the damage to be in the original pr, since Ryll's changes to embedded objects was a broad nerf to embed damage, and use whatever he suggests. Code comments suggest 55?

**Review by Twaticus — APPROVED** [review]

**Comment by necromanceranne** [comment]
Well I can't get Denton's attention. I guess I'll just consider a middle of the rode approach and ask whether 60 is appropriate, putting the weapon at it's closest approximate. The mosin.

**Comment by lolman360** [comment]
they're magspears, right? what about adding an action to the gun that magnetically yanks nearby spears spear (potentially out of people) and throws it at your face

**Comment by wesoda25** [comment]
60 damage still seems pretty nutted, but this shit looks so fun to use I can’t even complain.

What I CAN complain about though is you doubling the amount of ammo. This gun is gonna be a powerhouse and we should definitely try and limit the amount of spears.

**Comment by necromanceranne** [comment]
You're spending 20 tc on this gun and this gun only in an rng kit. As far as power levels go, a 40 shot weapon you cannot conceal is not too outrageous compared to some of the other bundles. If your entire ginmick is just shooting people with harpoons, having that stop being a thing after 20 something shots when you blew all your TC on it and cannot create more ammo seems ridiculous given it's now it's kind of an overpriced maint musket.

I'd rather get the bullet itself right then worry about longevity, since as far as kits go they're all pretty much going to get you killed fairly quickly either way since you basically are paying to get dunked by a roll of the dice and then dunked by extremely limited scope gear.

Also fuck it's conflicting with my own pr aaah QwQ

**Pull request merged**